Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -1023,7 +1071,14 @@ void Connection::dispatchSyncMessage(Decoder& decoder)
 
 void Connection::dispatchDidReceiveInvalidMessage(MessageName messageName)
 {
-    ensureOnMainRunLoop([this, protectedThis = Ref { *this }, messageName]() mutable {
+    if (isMainRunLoopConnection() && isMainRunLoop()) {
```

Comment:
This is a tricky sequence.
My initial thought is that the caller should ensure that the dispatchDidReceiveInvalidMessage is called in the IPC receive work queue. Currently the only caller dispatches, which is silly for Cocoa. I think also Unix runs in the receive work queue, not sure about windows. It *should*.

This could be done in a separate patch and tested in itself.